docs/user-guide/ingress-path-matching.md

```diff
@@ -83,7 +83,7 @@ spec:
 The ingress controller would define the following location blocks, in order of descending length, within the NGINX template for the `test.com` server:
```

Comment:
Why would it define more than single location block? Setting path to `/foo/bar/.+` means I require all paths to be prefixed with `/foo/bar/` and followed by at least a single character.

If in addition to above I want to match `/foo/bar` and `/foo/bar/` I'd define my as as `/foo/bar/?.*`

Reply:
`/foo/bar/?.*` wouldn't work because it would always catch requests to `/foo/bar/`, making that location inaccesible.

Because of the first match policy, what I have will work, but it makes more sense to have the following:

- `/foo/bar/$`
- `/foo/bar/`
- `/foo/bar`

That way it will always use the longest path match but check for `/foo/bar/$` first.

Reply:
but there's a concrete example above and it has just single path and backend service, so why not fix this?